tests/lib/rules/no-this-in-sfc.js

```diff
@@ -16,6 +16,8 @@ const ERROR_MESSAGE = 'Stateless functional components should not use this';
 const rule = require('../../../lib/rules/no-this-in-sfc');
 const RuleTester = require('eslint').RuleTester;
 
+require('babel-eslint');
```

Comment:
@ljharb do you know if this is really necessary? Not all of the test files that use this parser actually have the require.

Reply:
if you run those test files in isolation, do they pass?

Reply:
I don't see any differences after removing `require('babel-eslint')`. See the proof #2004 . Also removed it here.